### PR TITLE
stacked bar support for query builder results

### PR DESCRIPTION
**Pull request by eric-j**
https://new-relic.atlassian.net/browse/NR-262907

### CODE REVIEW

File: shared/ui/Stream/NRQL/utils.ts [hunk 1]
```diff
@@ -30,3 +30,40 @@ export function renameKeyToName(arr: NRQLResult[]): NRQLResult[] {
...
+	return str;
+};
+
+export const formatXAxisTime = time => {
+	const date = new Date(time * 1000);
+	return `${date.toLocaleTimeString()}`;
+};
+
+export const getUniqueDataKeyAndFacetValues = (results, facet) => {
+	const result = results ? results[0] : undefined;
```
Comment (CalvinAllen) [inline shared/ui/Stream/NRQL/utils.ts:46]: Could just have the fallback be `{}` here and then you wouldn't need it on line 51?

File: shared/ui/Stream/NRQL/utils.ts [hunk 2]
```diff
@@ -30,3 +30,40 @@ export function renameKeyToName(arr: NRQLResult[]): NRQLResult[] {
...
+	const defaultFilterKeys = ["beginTimeSeconds", "endTimeSeconds", "facet"];
+	const filterKeys = defaultFilterKeys.concat(facet);
+
+	const dataKeys = Object.keys(result || {}).filter(key => !filterKeys.includes(key));
+	const uniqueFacetValues: string[] = [...new Set<string>(results.map(obj => obj.facet))];
+	return { dataKeys, uniqueFacetValues };
+};
+
+export const fillNullValues = array => {
+	array.forEach((obj, i) => {
```
Comment (CalvinAllen) [inline shared/ui/Stream/NRQL/utils.ts:57]: Can `array` be undefined?

File: shared/ui/Stream/NRQL/utils.ts [hunk 3]
```diff
@@ -30,3 +30,40 @@ export function renameKeyToName(arr: NRQLResult[]): NRQLResult[] {
...
 }
+
+export const truncate = (str: string, max: number) => {
+	if (!str) return str;
+	if (str.length >= max) return `${str.substring(0, max - 1)}${"\u2026"}`;
+	return str;
+};
+
+export const formatXAxisTime = time => {
+	const date = new Date(time * 1000);
```
Comment (CalvinAllen) [inline shared/ui/Stream/NRQL/utils.ts:41]: Can `time` be undefined?

**Review by CalvinAllen — APPROVED** [review]
Just some general questions

File: shared/ui/Stream/NRQL/utils.ts (hunk 2 above)
Comment (eric-j) [reply shared/ui/Stream/NRQL/utils.ts:57]: ill add a check

File: shared/ui/Stream/NRQL/utils.ts (hunk 3 above)
Comment (eric-j) [reply shared/ui/Stream/NRQL/utils.ts:41]: ill add a check

**Pull request merged**